projects/dataset_recorder/auto_dataset_recorder_simulated.cc

```diff
@@ -0,0 +1,91 @@
+#include "auto_controller.h"
+
+// BoB robotics includes
+#include "common/macros.h"
+#include "navigation/image_database.h"
+#include "robots/ackermann/simulated_ackermann.h"
+#include "video/randominput.h"
+
+// Third-party includes
+#include "third_party/matplotlibcpp.h"
+
+// Standard C++ includes
+#include <vector>
+
+using namespace BoBRobotics;
+using namespace units::angle;
+using namespace units::length;
+using namespace units::velocity;
+using namespace units::literals;
+using namespace units::math;
+namespace plt = matplotlibcpp;
+
+class RobotArrow
```

Comment:
Why is this a class rather than just a function? ``plt::quiver`` is going to surely do more than enough memory allocating of its own to make any attempt to save allocations of ``m_X``, ``m_Y`` etc worthwhile. If you really care just have the vectors at ``bobMain`` scope and do this manually in the one place you actually draw an arrow!

Reply:
True true